### PR TITLE
ANA-688: Removing ComplexInstruments.cs entirely

### PR DESCRIPTION
**Pull request by jz-finbourne**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass
- [ ] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 1]
```diff
@@ -150,6 +150,60 @@ internal static LusidInstrument CreateExampleInterestRateSwap()
                 instrumentType: LusidInstrument.InstrumentTypeEnum.InterestRateSwap
             );
         }
+        
+        internal static InterestRateSwap CreateSwapByNamedConventions(DateTimeOffset startDate, DateTimeOffset maturityDate, decimal fixedRate, FlowConventionName flowConventionName, FlowConventionName indexConventionName, string fixedLegDirection = "Pay", decimal notional=100m)
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:154]: All of the other functions in this file have self explanatory names. This one is a bit less obvious, can you add a docstring to this one?

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/PortfolioCashFlows.cs [hunk 2]
```diff
@@ -384,8 +368,8 @@ public void LifeCycleManagementForCashSettledEquityOption()
 
             // GET all upsertable cashflows (transactions) for the EquityOption
             var allEquityOptionCashFlows = _transactionPortfoliosApi.GetUpsertablePortfolioCashFlows(
-                    _portfolioScope,
-                    _portfolioCode,
+                 _portfolioScope,
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/PortfolioCashFlows.cs:371]: Your indenting is off by a single space

File: sdk/Lusid.Sdk.Tests/tutorials/Ibor/PortfolioCashFlows.cs [hunk 3]
```diff
@@ -237,10 +225,6 @@ public void ExampleUpsertablePortfolioCashFlowsForFxForwards(bool isNdf)
         [Test]
         public void LifeCycleManagementForFxForward()
         {
-            // CREATE portfolio
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Ibor/PortfolioCashFlows.cs:240]: Can you remove the one from line 341 as well?

File: sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+using System.Linq;
+using Lusid.Sdk.Model;
+using Lusid.Sdk.Tests.Utilities;
+using LusidFeatures;
+using NUnit.Framework;
+
+namespace Lusid.Sdk.Tests.Tutorials.Instruments
+{
+    [TestFixture]
+    public class InterestRateSwapWithNamedConventions: DemoInstrumentBase
```
Comment (fightingsleep) [inline sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:12]: Whether you want to do it now or later, this class probably needs additional documentation. I don't think the meaning of "with named conventions" is necessarily obvious to everyone who reads it

Comment (jz-finbourne) [reply sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapWithNamedConventionsExamples.cs:12]: I copied the naming from existing file tbh. 
Most definitely be revisiting this add tests and touch up!

**Pull request merged**